### PR TITLE
Add uploader

### PR DESCRIPTION
**Pull request by Frozenka**
# Description

Addition of "uploader," a simply utility that allows easy file transfer to a remote OS, supporting both Windows and Linux. The utility will ask for the OS, port, and file you want to send. Once done, it opens a Python web server and injects the download command directly into your clipboard (example: wget http://{IP}/YOURFILE). Once the file is retrieved by the remote machine, the web server closes.

# Related issues

> /

# Point of attention

> Things you are not sure about that deserve special attention if you have doubts or questions.
Design sans titre

### CODE REVIEW

File: sources/install/package_base.sh [hunk 1]
```diff
@@ -396,6 +396,16 @@ function post_install() {
     if [ -d "/root/Desktop" ]; then rm -r /root/Desktop; fi
 }
 
+function uploader() {
```
Comment (QU35T-code) [inline sources/install/package_base.sh:399]: Missing uploader() function call (misc package ?)

File: sources/install/package_base.sh [hunk 2]
```diff
@@ -396,6 +396,16 @@ function post_install() {
...
+    colorecho "Installing Uploader"
+    git clone https://github.com/Frozenka/uploader.git /opt/tools/uploader
```
Comment (QU35T-code) [inline sources/install/package_base.sh:401]: Can you add `-C /opt/tools` and `--depth 1` ? (`git -C /opt/tools/ clone --depth 1 https://.....`)

File: sources/install/package_base.sh [hunk 3]
```diff
@@ -396,6 +396,16 @@ function post_install() {
...
+    pip3 install -r /opt/tools/uploader/requirements.txt
```
Comment (QU35T-code) [inline sources/install/package_base.sh:402]: You need to use `venv`. For exemple you can add a `setup.py` / `pyproject.toml` file in the root of your repo (example: https://github.com/p0dalirius/smbclient-ng) and use `pipx` to install your tool.
Or, you can use a python venv -> https://exegol.readthedocs.io/en/latest/community/contributors.html

File: sources/install/package_base.sh [hunk 4]
```diff
@@ -396,6 +396,16 @@ function post_install() {
...
+    add-aliases uploader
+    add-history uploader
+    add-test-command "uploader.py --help"
```
Comment (QU35T-code) [inline sources/install/package_base.sh:405]: You refer to `uploader.py` but your alias file refers to the `uploader` command.

**Review by QU35T-code — CHANGES_REQUESTED** [review]
Hey @Frozenka,

Thanks for this PR.

File: sources/install/package_base.sh (hunk 4 above)
Comment (Frozenka) [reply sources/install/package_base.sh]: done

File: sources/install/package_base.sh [hunk 5]
```diff
@@ -396,6 +396,20 @@ function post_install() {
     if [ -d "/root/Desktop" ]; then rm -r /root/Desktop; fi
 }
 
+function uploader() {
+    colorecho "Installing Uploader"
+    git -C /opt/tools/ clone --depth 1 https://github.com/Frozenka/uploader.git 
+    cd /opt/tools/uploader || exit
+    python3 -m venv --system-site-package ./venv
+    source ./venv/bin/activate
+    pip3 install -r requirements.txt
```
Comment (QU35T-code) [inline sources/install/package_base.sh:405]: Replace `pip3` with `pip`

File: sources/assets/shells/aliases.d/uploader [hunk 6]
```diff
@@ -0,0 +1 @@
+alias uploader='python3 /opt/tools/uploader/uploader.py'
```
Comment (QU35T-code) [inline sources/assets/shells/aliases.d/uploader:1]: You must use the python binary venv : `alias uploader='/opt/tools/uploader/venv/bin/python3 /opt/tools/uploader/uploader.py'`

**Review by QU35T-code — CHANGES_REQUESTED** [review]
Can you move your installation to the `package_misc` file ?
In addition, the function call is missing

File: sources/install/package_base.sh (hunk 5 above)
Comment (Frozenka) [reply sources/install/package_base.sh]: done :)

File: sources/assets/shells/aliases.d/uploader (hunk 6 above)
Comment (Frozenka) [reply sources/assets/shells/aliases.d/uploader]: done

File: sources/install/package_base.sh (hunk 1 above)
Comment (QU35T-code) [reply sources/install/package_base.sh]: It missing function call, you need to call your function in the package_misc function

File: sources/install/package_misc.sh [hunk 7]
```diff
@@ -185,6 +185,20 @@ function install_creds() {
...
+    deactivate
+    add-aliases uploader
+    add-history uploader
+    add-test-command "uploader --help"
+    add-to-list "uploader,https://github.com/Frozenka/uploader,Tool for quickly downloading files to a remote machine based on the target operating system"
+}
+
 # Package dedicated to offensive miscellaneous tools
 function package_misc() {
     set_env
```
Comment (QU35T-code) [inline sources/install/package_misc.sh:204]: function call x) ?

File: sources/install/package_base.sh (hunk 5 above)
Comment (QU35T-code) [reply sources/install/package_base.sh]: No

File: sources/install/common.sh [hunk 8]
```diff
@@ -122,7 +122,6 @@ function catch_and_retry() {
   criticalecho-noexit "Command failed definitively after $retries attempts."
   return 1
 }
-
```
Comment (QU35T-code) [inline sources/install/common.sh:125]: Can you revert this file ?

**Review by QU35T-code — CHANGES_REQUESTED** [review]

**Pull request merged**